core/rawdb/ancient_utils.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2022 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package rawdb
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethdb"
+)
+
+// freezerInfo contains the basic information of the freezer.
+type freezerInfo struct {
+	name  string                        // The identifier of freezer
+	head  uint64                        // The number of last stored item in the freezer
+	tail  uint64                        // The number of first stored item in the freezer
+	sizes map[string]common.StorageSize // The storage size per table
+}
+
+// count returns the number of stored items in the freezer.
+func (info freezerInfo) count() uint64 {
```

Comment:
Nitpick, please make this a method on a pointer, not the value. Passing things by value should IMHO be the occasional thing we explicitly desire vs the default.

Reply:
I.e. not just this method, all mtheods.